app/src/main/java/org/schabi/newpipe/info_list/InfoItemDialog.java

```diff
@@ -1,54 +1,296 @@
+import org.schabi.newpipe.util.StreamDialogDefaultEntry;
+import org.schabi.newpipe.util.StreamDialogEntry;
+import org.schabi.newpipe.util.external_communication.KoreUtils;
 
-public class InfoItemDialog {
-    private final AlertDialog dialog;
+import java.util.ArrayList;
+import java.util.List;
 
-    public InfoItemDialog(@NonNull final Activity activity,
-                          @NonNull final StreamInfoItem info,
-                          @NonNull final String[] commands,
-                          @NonNull final DialogInterface.OnClickListener actions) {
-        this(activity, commands, actions, info.getName(), info.getUploaderName());
-    }
+/**
+ * Dialog for a {@link StreamInfoItem}.
+ * The dialog'S content are actions that can be performed on the {@link StreamInfoItem}.
+ * This dialog is mostly used for longpress context menus.
+ */
+public final class InfoItemDialog {
+    /**
+     * Ideally, {@link InfoItemDialog} would extend {@link AlertDialog}.
+     * However, extending {@link AlertDialog} requires many additional lines
+     * and brings more complexity to this class, especially the constructor.
+     * To circumvent this, an {@link AlertDialog.Builder} is used in the constructor.
+     * Its result is stored in this class variable to allow access via the {@link #show()} method.
+     */
+    private final AlertDialog dialog;
 
-    public InfoItemDialog(@NonNull final Activity activity,
-                          @NonNull final String[] commands,
-                          @NonNull final DialogInterface.OnClickListener actions,
-                          @NonNull final String title,
-                          @Nullable final String additionalDetail) {
+    private InfoItemDialog(@NonNull final Activity activity,
+                           @NonNull final Fragment fragment,
+                           @NonNull final StreamInfoItem info,
+                           @NonNull final List<StreamDialogEntry> entries) {
 
         final View bannerView = View.inflate(activity, R.layout.dialog_title, null);
         bannerView.setSelected(true);
 
         final TextView titleView = bannerView.findViewById(R.id.itemTitleView);
-        titleView.setText(title);
+        titleView.setText(info.getName());
 
         final TextView detailsView = bannerView.findViewById(R.id.itemAdditionalDetails);
-        if (additionalDetail != null) {
-            detailsView.setText(additionalDetail);
+        if (info.getUploaderName() != null) {
+            detailsView.setText(info.getUploaderName());
             detailsView.setVisibility(View.VISIBLE);
         } else {
             detailsView.setVisibility(View.GONE);
         }
 
+        final String[] items = entries.stream()
+                .map(entry -> entry.getString(activity)).toArray(String[]::new);
+
+        final DialogInterface.OnClickListener action = (d, index) ->
+            entries.get(index).action.onClick(fragment, info);
+
```

Comment:
optional: I think those could also be done inline.

Reply:
I did not do this, because I wanted to increase the code's readability. If that is done inline, one needs way more time to understand what's going on.

Reply:
I don't think so.
If you use line breaks and/or add a comment I see no problem there.

For me find one's way inside bloated java-files that has useless variable assignments (*cough* Player.java *cough*) is a way bigger problem.